### PR TITLE
[new] - Add accessible-emoji rule.

### PR DESCRIPTION
**Pull request by evcohen**
Read [this article](http://tink.uk/accessible-emoji/) from @LJWatson and thought we could add a rule to enforce this

### CODE REVIEW

File: docs/rules/accessible-emoji.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#### Resources
+1. [Lèonie Watson](http://tink.uk/accessible-emoji/)
+
+## Rule details
+
+This rule takes no arguments.
+
+### Succeed
+```jsx
+<span role="img" aria-label="Snowman">&#9731;</span>
```
Comment (jessebeach) [inline docs/rules/accessible-emoji.md]: Add an `aria-labelledby` example.

File: __tests__/src/rules/accessible-emoji-test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+ruleTester.run('accessible-emoji', rule, {
+  valid: [
+    { code: '<div />;', parserOptions },
+    { code: '<span />', parserOptions },
+    { code: '<span>No emoji here!</span>', parserOptions },
+    { code: '<span role="img" aria-label="Panda face">🐼</span>', parserOptions },
+    { code: '<span role="img" aria-label="Snowman">&#9731;</span>', parserOptions },
+    { code: '<span>{props.emoji}</span>', parserOptions },
```
Comment (jessebeach) [inline __tests__/src/rules/accessible-emoji-test.js]: Add an aria-labelledby example.

**Comment by ljharb** [comment]
It'd be great if this could be rebased, so as to remove all merge commits.

**Comment by jessebeach** [comment]
Rebased on master and I changed the find method to return a boolean.

**Comment by jessebeach** [comment]
Great new rule, thanks!

**Pull request merged**